afs-cassandra/src/main/java/com/powsybl/afs/cassandra/CassandraAppStorage.java

```diff
@@ -799,22 +800,36 @@ private UUID deleteNode(UUID nodeUuid) {
         clearTimeSeries(nodeUuid, statements);
 
         // dependencies
+        List<Pair<UUID, String>> dependencies = getDependencyInfo(nodeUuid);
         ResultSet resultSet = getSession().execute(select(NAME, FROM_ID).from(BACKWARD_DEPENDENCIES)
                                                                                 .where(eq(TO_ID, nodeUuid)));
+
+        List<Pair<UUID, String>> backwardDependencies = new ArrayList<>();
```

Comment:
Why using a list of pair? Is it a Map?

Reply:
in this case, it could be a Map<String, List<UUID>> since this node can be a dependency of multiple other nodes with the same dependency link name, or (since not likely to happen) linked in different ways (dependency name) to the same object.
that's why i went to use a List<Pair<UUID,String>> in both this case for homogeneity.